monkey/monkey_island/cc/resources/pba_file_upload.py

```diff
@@ -102,11 +101,15 @@ def _update_config(self, target_os: str, safe_filename: str):
         agent_configuration = self._agent_configuration_repository.get_configuration()
 
         if target_os == LINUX_PBA_TYPE:
-            custom_pbas = replace(agent_configuration.custom_pbas, linux_filename=safe_filename)
+            custom_pbas = agent_configuration.custom_pbas.copy(
+                update={"linux_filename": safe_filename}
+            )
         else:
-            custom_pbas = replace(agent_configuration.custom_pbas, windows_filename=safe_filename)
+            custom_pbas = agent_configuration.custom_pbas.copy(
+                update={"windows_filename": safe_filename}
+            )
 
-        updated_agent_configuration = replace(agent_configuration, custom_pbas=custom_pbas)
+        updated_agent_configuration = agent_configuration.copy(update={"custom_pbas": custom_pbas})
```

Comment:
Any reason to not just modify `agent_configuration` directly? Since these objects are now mutable, I don't see any need to make copies.